internal/blocktx/processor_helpers.go

```diff
@@ -140,3 +121,29 @@ func progressIndices(total, steps int) map[int]int {
 	progress[total] = 100
 	return progress
 }
+
+func withRetry(ctx context.Context, fn func() error, n int) error {
+	n = int(math.Max(float64(n), 1)) // ensure n is at least 1
+	var rerr error
+
+	for i := range n {
+		select {
+		case <-ctx.Done():
+			rerr = errors.Join(rerr, fmt.Errorf("failure on %d try: %w", i+1, ctx.Err()))
+			return rerr
+
+		default:
+			err := fn()
+
+			// return if the function completes successfully
+			if err == nil {
+				return nil
+			}
+
+			rerr = errors.Join(rerr, fmt.Errorf("failure on %d try: %w", i+1, err))
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
```

Comment:
This could be done similarly as the other retry function

https://github.com/bitcoin-sv/arc/blob/2b723ddcdab35fb753041ae700177376609f0f3e/internal/test_utils/utils.go#L57:L70

But this is also fine